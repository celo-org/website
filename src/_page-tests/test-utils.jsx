import { render } from "@testing-library/react";
import * as React from "react";
import i18nForTests from "src/utils/i18nForTests";
import { I18nextProvider } from "react-i18next";
export function TestProvider({ children }) {
    return <I18nextProvider i18n={i18nForTests}>{children}</I18nextProvider>;
}
export const renderIgnoringUnstableFlushDiscreteUpdates = (component) => {
    const originalError = console.error;
    const error = jest.fn();
    console.error = error;
    const result = render(component);
    expect(error).toHaveBeenCalledTimes(1);
    expect(error).toHaveBeenCalledWith("Warning: unstable_flushDiscreteUpdates: Cannot flush updates when React is already rendering.%s", expect.any(String));
    console.error = originalError;
    return result;
};
//# sourceMappingURL=test-utils.jsx.map