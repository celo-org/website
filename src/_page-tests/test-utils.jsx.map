{"version":3,"file":"test-utils.jsx","sourceRoot":"","sources":["test-utils.tsx"],"names":[],"mappings":"AAAA,OAAO,EAAE,MAAM,EAAE,MAAM,wBAAwB,CAAA;AAC/C,OAAO,KAAK,KAAK,MAAM,OAAO,CAAA;AAC9B,OAAO,YAAY,MAAM,wBAAwB,CAAA;AAIjD,OAAO,EAAE,eAAe,EAAE,MAAM,eAAe,CAAA;AAI/C,MAAM,UAAU,YAAY,CAAC,EAAE,QAAQ,EAAS;IAC9C,OAAO,CAAC,eAAe,CAAC,IAAI,CAAC,CAAC,YAAY,CAAC,CAAC,CAAC,QAAQ,CAAC,EAAE,eAAe,CAAC,CAAA;AAC1E,CAAC;AAID,MAAM,CAAC,MAAM,0CAA0C,GAAG,CAAC,SAA6B,EAAE,EAAE;IAE1F,MAAM,aAAa,GAAG,OAAO,CAAC,KAAK,CAAA;IACnC,MAAM,KAAK,GAAG,IAAI,CAAC,EAAE,EAAE,CAAA;IACvB,OAAO,CAAC,KAAK,GAAG,KAAK,CAAA;IACrB,MAAM,MAAM,GAAG,MAAM,CAAC,SAAS,CAAC,CAAA;IAChC,MAAM,CAAC,KAAK,CAAC,CAAC,qBAAqB,CAAC,CAAC,CAAC,CAAA;IACtC,MAAM,CAAC,KAAK,CAAC,CAAC,oBAAoB,CAChC,iGAAiG,EACjG,MAAM,CAAC,GAAG,CAAC,MAAM,CAAC,CACnB,CAAA;IACD,OAAO,CAAC,KAAK,GAAG,aAAa,CAAA;IAE7B,OAAO,MAAM,CAAA;AACf,CAAC,CAAA"}