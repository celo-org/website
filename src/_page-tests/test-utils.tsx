import { render } from "@testing-library/react"
import * as React from "react"
import i18nForTests from "src/utils/i18nForTests"
interface Props {
  children?: React.ReactNode
}
import { I18nextProvider } from "react-i18next"



export function TestProvider({ children }: Props) {
  return <I18nextProvider i18n={i18nForTests}>{children}</I18nextProvider>
}

// https://github.com/testing-library/react-testing-library/issues/470
// an issue with muted on video causing a rerender, this seems to be be best solution for now
export const renderIgnoringUnstableFlushDiscreteUpdates = (component: React.ReactElement) => {
  // tslint:disable: no-console
  const originalError = console.error
  const error = jest.fn()
  console.error = error
  const result = render(component)
  expect(error).toHaveBeenCalledTimes(1)
  expect(error).toHaveBeenCalledWith(
    "Warning: unstable_flushDiscreteUpdates: Cannot flush updates when React is already rendering.%s",
    expect.any(String)
  )
  console.error = originalError
  // tslint:enable: no-console
  return result
}
