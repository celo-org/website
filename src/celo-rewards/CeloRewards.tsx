import * as React from "react"
import { StyleSheet, Text, View } from "react-native"
import OpenGraph from "src/header/OpenGraph"
import { NameSpaces, useTranslation } from "src/i18n"
import Link from "src/shared/Link"
import menuItems, { pagePaths } from "src/shared/menu-items"
import { colors, typeFaces } from "src/styles"
import AddCusdButton from "./AddCusdButton"
import RewardsKickOff from "./RewardsKickOff"
import TierList from "./TierList"
import TitleAndDescription from "./TitleAndDescription"

function CeloRewards({}) {
  const { t } = useTranslation(NameSpaces.celoRewards)

  return (
    <>
      <OpenGraph
        title={t("title")}
        path={menuItems.CELO_REWARDS.link}
        description={t("description.first")}
      />
      <View style={styles.container}>
        <TitleAndDescription
          title={t("title")}
          description={
            <>
              {t("description.first")}
              {"\n"}
              <TierList baseTranslation="description" totalTiers={3} />
              {"\n"}
              {t("description.second")}
            </>
          }
        />
        <RewardsKickOff />
        <AddCusdButton />
        <Link href={pagePaths.CELO_REWARDS_TERMS.link} passHref={true}>
          <Text accessibilityRole="link" style={styles.learnMore} testID="learnMore">
            {t("learnMore")}
          </Text>
        </Link>
      </View>
    </>
  )
}

CeloRewards.getInitialProps = async () => ({
  namespacesRequired: [NameSpaces.celoRewards],
})

export default CeloRewards

const styles = StyleSheet.create({
  container: {
    marginTop: 20,
  },
  learnMore: {
    display: "flex",
    justifyContent: "center",
    fontFamily: typeFaces.futura,
    fontSize: 17,
    fontWeight: "500",
    marginBottom: 20,
    color: colors.greenUI,
  },
})
