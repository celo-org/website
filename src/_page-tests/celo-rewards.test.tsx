import CeloRewards from "pages/celo-rewards"
import * as React from "react"
import * as renderer from "react-test-renderer"
import { TestProvider } from "src/_page-tests/test-utils"

describe("CeloRewards", () => {
  it("renders", () => {
    const tree = renderer
      .create(
        <TestProvider>
          <CeloRewards />
        </TestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
