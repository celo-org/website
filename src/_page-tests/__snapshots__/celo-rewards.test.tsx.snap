// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CeloRewards renders 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "marginTop": "20px",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignSelf": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexDirection": "column",
        "alignSelf": "center",
        "flexDirection": "column",
        "maxWidth": "100vw",
        "msFlexDirection": "column",
        "msFlexItemAlign": "center",
        "msGridRowAlign": "center",
        "paddingLeft": "10px",
        "paddingRight": "10px",
        "width": "100%",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitBoxFlex": 0,
          "WebkitFlexGrow": 0,
          "WebkitFlexShrink": 0,
          "flexGrow": 0,
          "flexShrink": 0,
          "msFlexNegative": 0,
          "msFlexPositive": 0,
          "paddingBottom": "10px",
          "paddingLeft": "10px",
          "paddingRight": "10px",
          "paddingTop": "10px",
          "width": "100%",
        }
      }
    >
      <h1
        aria-level="1"
        className="css-reset-4rbku5 css-text-901oao"
        dir="auto"
        onClick={[Function]}
        role="heading"
        style={
          Object {
            "color": "rgba(46,51,56,1.00)",
            "fontFamily": "Jost, futura-pt, futura, sans-serif",
            "fontSize": "26px",
            "fontWeight": "500",
            "lineHeight": "32px",
            "marginBottom": "16px",
            "textRendering": "geometricPrecision",
            "textTransform": "none",
          }
        }
      >
        CELO Rewards Program
      </h1>
      <div
        className="css-text-901oao"
        dir="auto"
        onClick={[Function]}
        style={
          Object {
            "color": "rgba(46,51,56,1.00)",
            "fontFamily": "Jost, futura-pt, futura, sans-serif",
            "fontSize": "18px",
            "fontWeight": "400",
            "lineHeight": "24px",
            "textRendering": "geometricPrecision",
          }
        }
      >
        Confirmed cUSD wallets can earn rewards by holding a minimum monthly average balance of:
        

        <ul
          className="css-reset-4rbku5 css-view-1dbjc4n"
          role="list"
          style={
            Object {
              "display": "inline-flex",
              "listStyle": "disc",
              "marginTop": "20px",
              "paddingLeft": "20px",
            }
          }
        >
          <li
            className="css-reset-4rbku5 css-text-901oao css-textHasAncestor-16my406"
            onClick={[Function]}
            role="listitem"
            style={
              Object {
                "color": "rgba(46,51,56,1.00)",
                "display": "list-item",
                "fontFamily": "EB Garamond, eb-garamond, Garamond, serif",
                "fontSize": "20px",
                "lineHeight": "28px",
                "listStyle": "inherit",
                "marginBottom": "20px",
                "textRendering": "geometricPrecision",
              }
            }
          >
            <span
              className="css-text-901oao css-textHasAncestor-16my406"
              onClick={[Function]}
              style={
                Object {
                  "color": "rgba(46,51,56,1.00)",
                  "fontFamily": "Jost, futura-pt, futura, sans-serif",
                  "fontSize": "18px",
                  "fontWeight": "400",
                  "lineHeight": "24px",
                  "textRendering": "geometricPrecision",
                }
              }
            >
              <span
                className="css-text-901oao css-textHasAncestor-16my406"
                onClick={[Function]}
                style={
                  Object {
                    "color": "rgba(46,51,56,1.00)",
                    "fontFamily": "Jost, futura-pt, futura, sans-serif",
                    "fontSize": "18px",
                    "fontWeight": "bold",
                    "lineHeight": "24px",
                    "textRendering": "geometricPrecision",
                  }
                }
              >
                10 cUSD
              </span>
              

              Earn $1 worth of CELO per month
            </span>
          </li>
          <li
            className="css-reset-4rbku5 css-text-901oao css-textHasAncestor-16my406"
            onClick={[Function]}
            role="listitem"
            style={
              Object {
                "color": "rgba(46,51,56,1.00)",
                "display": "list-item",
                "fontFamily": "EB Garamond, eb-garamond, Garamond, serif",
                "fontSize": "20px",
                "lineHeight": "28px",
                "listStyle": "inherit",
                "marginBottom": "20px",
                "textRendering": "geometricPrecision",
              }
            }
          >
            <span
              className="css-text-901oao css-textHasAncestor-16my406"
              onClick={[Function]}
              style={
                Object {
                  "color": "rgba(46,51,56,1.00)",
                  "fontFamily": "Jost, futura-pt, futura, sans-serif",
                  "fontSize": "18px",
                  "fontWeight": "400",
                  "lineHeight": "24px",
                  "textRendering": "geometricPrecision",
                }
              }
            >
              <span
                className="css-text-901oao css-textHasAncestor-16my406"
                onClick={[Function]}
                style={
                  Object {
                    "color": "rgba(46,51,56,1.00)",
                    "fontFamily": "Jost, futura-pt, futura, sans-serif",
                    "fontSize": "18px",
                    "fontWeight": "bold",
                    "lineHeight": "24px",
                    "textRendering": "geometricPrecision",
                  }
                }
              >
                100 cUSD
              </span>
              

              Earn $5 worth of CELO per month
            </span>
          </li>
          <li
            className="css-reset-4rbku5 css-text-901oao css-textHasAncestor-16my406"
            onClick={[Function]}
            role="listitem"
            style={
              Object {
                "color": "rgba(46,51,56,1.00)",
                "display": "list-item",
                "fontFamily": "EB Garamond, eb-garamond, Garamond, serif",
                "fontSize": "20px",
                "lineHeight": "28px",
                "listStyle": "inherit",
                "marginBottom": "20px",
                "textRendering": "geometricPrecision",
              }
            }
          >
            <span
              className="css-text-901oao css-textHasAncestor-16my406"
              onClick={[Function]}
              style={
                Object {
                  "color": "rgba(46,51,56,1.00)",
                  "fontFamily": "Jost, futura-pt, futura, sans-serif",
                  "fontSize": "18px",
                  "fontWeight": "400",
                  "lineHeight": "24px",
                  "textRendering": "geometricPrecision",
                }
              }
            >
              <span
                className="css-text-901oao css-textHasAncestor-16my406"
                onClick={[Function]}
                style={
                  Object {
                    "color": "rgba(46,51,56,1.00)",
                    "fontFamily": "Jost, futura-pt, futura, sans-serif",
                    "fontSize": "18px",
                    "fontWeight": "bold",
                    "lineHeight": "24px",
                    "textRendering": "geometricPrecision",
                  }
                }
              >
                500 cUSD
              </span>
              

              Earn $10 worth of CELO per month
            </span>
          </li>
        </ul>
        

        Rewards are credited to your walletâ€™s CELO balance at the beginning of each month until the program ends.
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignSelf": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexDirection": "column",
        "alignSelf": "center",
        "flexDirection": "column",
        "maxWidth": "100vw",
        "msFlexDirection": "column",
        "msFlexItemAlign": "center",
        "msGridRowAlign": "center",
        "paddingLeft": "10px",
        "paddingRight": "10px",
        "width": "100%",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitBoxFlex": 0,
          "WebkitFlexGrow": 0,
          "WebkitFlexShrink": 0,
          "flexGrow": 0,
          "flexShrink": 0,
          "msFlexNegative": 0,
          "msFlexPositive": 0,
          "paddingBottom": "10px",
          "paddingLeft": "10px",
          "paddingRight": "10px",
          "paddingTop": "10px",
          "width": "100%",
        }
      }
    >
      <div
        className="css-text-901oao"
        dir="auto"
        onClick={[Function]}
        style={
          Object {
            "color": "rgba(46,51,56,1.00)",
            "fontFamily": "Jost, futura-pt, futura, sans-serif",
            "fontSize": "18px",
            "fontWeight": "400",
            "lineHeight": "24px",
            "textRendering": "geometricPrecision",
          }
        }
      >
        <span
          className="css-text-901oao css-textHasAncestor-16my406"
          onClick={[Function]}
          style={
            Object {
              "color": "rgba(46,51,56,1.00)",
              "fontFamily": "Jost, futura-pt, futura, sans-serif",
              "fontSize": "18px",
              "fontWeight": "bold",
              "lineHeight": "24px",
              "textRendering": "geometricPrecision",
            }
          }
        >
          Rewards Kickoff:
        </span>
         
        CELO rewards are 4x until March 1st! Expect to receive weekly rewards.
      </div>
    </div>
  </div>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao"
    data-testid="AddCusd"
    dir="auto"
    href="celo://wallet/cashIn"
    onClick={[Function]}
    onMouseEnter={[Function]}
    role="link"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxPack": "center",
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "backgroundColor": "rgba(26,183,117,1.00)",
        "borderBottomLeftRadius": "24px",
        "borderBottomRightRadius": "24px",
        "borderBottomWidth": "0px",
        "borderLeftWidth": "0px",
        "borderRightWidth": "0px",
        "borderTopLeftRadius": "24px",
        "borderTopRightRadius": "24px",
        "borderTopWidth": "0px",
        "color": "rgba(255,255,255,1.00)",
        "cursor": "pointer",
        "display": "flex",
        "fontFamily": "Jost, futura-pt, futura, sans-serif",
        "fontSize": "17px",
        "fontWeight": "600",
        "height": "48px",
        "justifyContent": "center",
        "lineHeight": "16px",
        "marginBottom": "30px",
        "marginLeft": "20px",
        "marginRight": "20px",
        "marginTop": "30px",
        "msFlexAlign": "center",
        "msFlexPack": "center",
        "textAlign": "center",
        "textRendering": "geometricPrecision",
      }
    }
  >
    Add cUSD
  </a>
  <a
    className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao"
    data-testid="learnMore"
    dir="auto"
    href="/celo-rewards-terms-and-conditions"
    onClick={[Function]}
    onMouseEnter={[Function]}
    role="link"
    style={
      Object {
        "WebkitBoxPack": "center",
        "WebkitJustifyContent": "center",
        "color": "rgba(26,183,117,1.00)",
        "display": "flex",
        "fontFamily": "Jost, futura-pt, futura, sans-serif",
        "fontSize": "17px",
        "fontWeight": "500",
        "justifyContent": "center",
        "marginBottom": "20px",
        "msFlexPack": "center",
      }
    }
  >
    Learn More
  </a>
</div>
`;
